Headless renderer won't compile with clang

I'm getting the following error when building with cmake. If I disable the warning then the compiler is not detected as CXX11 or CXX14 so no compilation occurs.

```
/root/Cinder/src/cinder/app/linux/RendererGlLinuxHeadless.cpp:88:38: fatal error: non-constant-expression cannot be narrowed from type 'int' to 'size_type' (aka 'unsigned long') in initializer list [-Wc++11-narrowing]
                std::vector<EGLDeviceEXT> devices{ numDevices };
                                                   ^~~~~~~~~~
/root/Cinder/src/cinder/app/linux/RendererGlLinuxHeadless.cpp:88:38: note: insert an explicit cast to silence this issue
                std::vector<EGLDeviceEXT> devices{ numDevices };
                                                   ^~~~~~~~~~
                                                   static_cast<size_type>( )
1 error generated.
```
https://github.com/cinder/Cinder/blob/d40e9a5c3218ee9a8127d100aaefcfcc7335c606/src/cinder/app/linux/RendererGlLinuxHeadless.cpp#L88

The full command I'm compiling  with is:

` cmake ..  -DCINDER_HEADLESS_GL=egl -DCINDER_DISABLE_ANTTWEAKBAR=1 -DCINDER_DISABLE_AUTO=1 -DCINDER_DISABLE_VIDEO=1 -DCINDER_BOOST_USE_SYSTEM=1 -DCMAKE_BUILD_TYPE=Release `

--- a/src/cinder/app/linux/RendererGlLinuxHeadless.cpp
+++ b/src/cinder/app/linux/RendererGlLinuxHeadless.cpp
@@ -85,7 +85,7 @@ bool RendererGlLinux::initialize( ci::ivec2 renderSize, RendererRef sharedRender
 		if( ! eglQueryDevicesEXT || ! eglQueryDevicesEXT( 0, nullptr, &numDevices ) || numDevices < 1 )
 			return false;
 
-		std::vector<EGLDeviceEXT> devices{ numDevices };
+		std::vector<EGLDeviceEXT> devices( numDevices );
 		if( ! eglQueryDevicesEXT( numDevices, devices.data(), &numDevices ) || numDevices < 1 )
 			return false;
 

